presto-parquet/src/main/java/com/facebook/presto/parquet/ColumnReaderFactory.java

```diff
@@ -63,8 +63,8 @@ private ColumnReaderFactory()
 
     public static ColumnReader createReader(RichColumnDescriptor descriptor, boolean batchReadEnabled)
     {
-        if (batchReadEnabled) {
-            final boolean isNested = descriptor.getPath().length > 1;
+        final boolean isNested = descriptor.getPath().length > 1;
+        if (batchReadEnabled && (!isNested || !isDecimalType(descriptor))) {
```

Comment:
I found this more straightforward to read.

```suggestion
        if (batchReadEnabled && (!(isNested && isDecimalType(descriptor)))) {
```